shared/modules/transaction.utils.ts

```diff
@@ -40,14 +37,15 @@ const erc1155Interface = new Interface(abiERC1155);
  * Determines if the maxFeePerGas and maxPriorityFeePerGas fields are supplied
  * and valid inputs. This will return false for non hex string inputs.
  *
- * @param {import('@metamask/transaction-controller').TransactionMeta} transaction -
- *  the transaction to check
- * @returns {boolean} true if transaction uses valid EIP1559 fields
+ * @param transactionMeta - the transaction to check
+ * @returns boolean true if transaction uses valid EIP1559 fields
  */
-export function isEIP1559Transaction(transaction) {
+export function isEIP1559Transaction(
+  transactionMeta: TransactionMeta,
+): boolean {
   return (
-    isHexString(transaction?.txParams?.maxFeePerGas) &&
-    isHexString(transaction?.txParams?.maxPriorityFeePerGas)
+    isHexString(transactionMeta?.txParams?.maxFeePerGas as string) &&
+    isHexString(transactionMeta?.txParams?.maxPriorityFeePerGas as string)
```

Comment:
We could enforce that these values are defined by providing safe fallback values that won't pollute their type signatures.

```suggestion
    isHexString(transactionMeta?.txParams?.maxFeePerGas ?? '') &&
    isHexString(transactionMeta?.txParams?.maxPriorityFeePerGas ?? '')
```

This lets us remove the `| undefined` member from these types instead of positively enforcing that they are of a certain type, which is much more dangerous.

Reply:
[5d8c13](https://github.com/MetaMask/metamask-extension/pull/23276/commits/5d8c1347481a2e3ad2be7f42b3b93e7eb80b9ffe)